app/src/androidTest/java/com/android/sample/endToend/MLEndToEndTest.kt

```diff
@@ -0,0 +1,120 @@
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.rule.GrantPermissionRule
+import com.android.sample.MainActivity
+import com.android.sample.mlUtils.CoachFeedback
+import com.android.sample.mlUtils.ExerciseFeedBackUnit
+import com.android.sample.mlUtils.exercisesCriterions.PlankExerciseCriterions
+import com.android.sample.model.camera.CameraViewModel
+import com.android.sample.ui.navigation.Route
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+
+class MLEndToEndTest {
+  private lateinit var mockCameraViewModel: CameraViewModel
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @get:Rule
+  val grantPermissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(
+          Manifest.permission.ACCESS_COARSE_LOCATION,
+          Manifest.permission.ACCESS_FINE_LOCATION,
+          Manifest.permission.POST_NOTIFICATIONS,
+          Manifest.permission.CAMERA)
+
+  @Before
+  fun setUp() {
+    val intent = Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java)
+    intent.putExtra("START_DESTINATION", Route.MAIN)
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+    context.startActivity(intent)
+    mockCameraViewModel = mock()
+    `when`(mockCameraViewModel.feedback)
+        .thenReturn(
+            listOf(
+                CoachFeedback(
+                    emptySet(),
+                    100f,
+                    10,
+                    ExerciseFeedBackUnit.SECONDS,
+                    true,
+                    PlankExerciseCriterions)))
+  }
+  /** This test is an end-to-end test that tests the ML flow of the app. */
+  @OptIn(ExperimentalTestApi::class)
+  @Test
+  fun mlEndToEndTest() {
+    // Starting from the main screen, go to the learnings screen
+    composeTestRule.onNodeWithTag("Video").assertIsDisplayed().performClick()
+    // We need to wait for the video to load and animation to end
+    composeTestRule.waitUntilExactlyOneExists(
+        matcher = hasTestTag("coachText"), timeoutMillis = 5000)
+    // Check if the coach text and button are displayed
+    composeTestRule.onNodeWithTag("coachText").assertIsDisplayed()
+    // Go to the ML screen
+    composeTestRule.onNodeWithTag("coachButton").assertIsDisplayed().performClick()
+    // Plank is selected by default, so we can just press the start button
+    composeTestRule.onNodeWithTag("saveButton").assertIsDisplayed().performClick()
+    // Press the record button twice to simulate a good plank
+    composeTestRule
+        .onNodeWithTag("recordButton")
+        .assertIsDisplayed()
+        .performClick()
+        .performClick()
+        .performClick()
+
+    // From now on, we just check that everything is correctly displayed according to the feedback
+    // given by the coach, which is gonna be a not detected
```

Comment:
Sorry could you please go over the phrasing of this comment, it is a bit weird formulated